src/libasr/pass/replace_symbolic.cpp

```diff
@@ -985,7 +997,9 @@ class ReplaceSymbolicVisitor : public PassUtils::PassVisitor<ReplaceSymbolicVisi
                     ASRUtils::EXPR(ASR::make_Var_t(al, x.base.base.loc, symbol))));
             }
             symbolic_vars_to_free.clear();
-            pass_result.push_back(al, ASRUtils::STMT(ASR::make_Return_t(al, x.base.base.loc)));
+            for (ASR::symbol_t* symbol : symbolic_vars_to_omit) {
+                symbolic_vars_to_free.insert(symbol);
+            }
```

Comment:
This code right here makes https://github.com/lcompilers/lpython/issues/2552 fail.